src/pages/scheduler/components/SchedulerCalendar.tsx

```diff
@@ -149,12 +150,26 @@ export const SchedulerCalendar = ({ term, courseCalendarEvents = [] }: Scheduler
         // if event does not have a scheduled time, move on.
         if (calendarEvent.meetingTime.time.indexOf('TBA') !== -1) return;
 
+        if (
+          calendarEvent.meetingTime.time.split(`- `)[1].split(` `)[1] === `pm` &&
+          parseInt(calendarEvent.meetingTime.time.split(`- `)[1].split(':')[0]) !== 12
+        ) {
+          let theTime = parseInt(calendarEvent.meetingTime.time.split(`- `)[1].split(`:`)[0]) + 12;
+          if (theTime > latestHour && theTime < 24) {
+            setLatestHour(theTime + 1);
+          }
+        }
+
+        const startEndDates = calendarEvent.meetingTime.dateRange.split('-');
+
+        const courseStartDate = new Date(startEndDates[0].replace(',', '') + ' 00:00:00 GMT');
+
         // check cache, if it exists, use it otherwise parse and set value in cache
         if (!EVENTS_CACHE[key]) {
           EVENTS_CACHE[key] = parseMeetingTimes(calendarEvent);
         }
 
-        const { lower: ruleLower, upper: ruleUpper, startDate: courseStartDate } = EVENTS_CACHE[key];
+        const { lower: ruleLower, upper: ruleUpper } = EVENTS_CACHE[key];
```

Comment:
Did `startDate` just get removed since it's not used?

Reply:
Its implemented again, and the redundant startDate declarations are being removed